src/sentry/integrations/slack/link_team.py

```diff
@@ -0,0 +1,198 @@
+                    "teams": teams,
+                    "channel_name": channel_name,
+                    "provider": integration.get_provider(),
+                },
+            )
+
+        team = Team.objects.get(id=team_id)
+        INSUFFICIENT_ROLE_MESSAGE = {
+            "heading": "Insufficient role",
+            "body": f"You must be an admin or higher and a member of the {team.slug} team in your Sentry organization to link teams.",
+        }
+        ALREADY_LINKED_MESSAGE = {
+            "heading": "Already linked",
+            "body": f"The {team.slug} team has already been linked to a Slack channel.",
+        }
+        SUCCESS_LINKED_MESSAGE = {
+            "heading": "Team linked",
+            "body": f"The {team.slug} team will now receive issue alert notifications in the {channel_name} channel.",
+        }
+        organization = Organization.objects.get_for_team_ids([team.id])[0]
+        identity = self.get_identity(integration, params["slack_id"])
+        org_member = OrganizationMember.objects.get(user=identity.user, organization=organization)
+        client = SlackClient()
+        token = (
+            integration.metadata.get("user_access_token") or integration.metadata["access_token"]
+        )
+        # if the org has open membership and the user is an admin or above
+        # OR if closed, ensure user is admin AND member of the team
+        if not (
+            organization.flags.allow_joinleave and org_member.role in ["admin", "manager", "owner"]
+        ) or (org_member.role in ["admin", "manager", "owner"] and team in [org_member.teams]):
+            return self.send_slack_message(
+                request,
+                client,
+                token,
+                INSUFFICIENT_ROLE_MESSAGE,
+                channel_id,
+                integration,
+            )
+
+        already_linked = ExternalActor.objects.filter(
+            actor_id=team.actor_id,
+            organization=organization,
+            integration=integration,
+            provider=ExternalProviders.SLACK.value,
+        )
+        if already_linked:
+            return self.send_slack_message(
+                request,
+                client,
+                token,
+                ALREADY_LINKED_MESSAGE,
+                channel_id,
+                integration,
+            )
+
+        external_team, created = ExternalActor.objects.get_or_create(
+            actor_id=team.actor_id,
+            organization=organization,
+            integration=integration,
+            provider=ExternalProviders.SLACK.value,
+            external_name=channel_name,
+            external_id=channel_id,
+        )
+
+        if created:
+            # turn on notifications for all of a team's projects
+            # TODO(ceo): Update this when API-1951 has been merged
+            team_projects = team.get_projects()
+            for project in team_projects:
+                NotificationSetting.objects.update_settings(
```

Comment:
this needs to be updated when API-1951 is merged

Reply:
I just merged the PR.